Mathlib/Data/Finsupp/Option.lean

```diff
@@ -9,12 +9,23 @@ public import Mathlib.Data.Finsupp.Basic
 public import Mathlib.Algebra.Module.Defs
 
 /-!
-# Declarations about finitely supported functions whose support is an `Option` type p
+# Operations on `Finsupp`s with an `Option` domain
+
+Similar to how `Finsupp.cons` and `Finsupp.tail` construct
+an object of type `Fin (n + 1) →₀ M` from a map `Fin n →₀ M` and vice versa,
+we define `Finsupp.optionElim'` and `Finsupp.some`
+to construct `Option α →₀ M` from a map α →₀ M, and vice versa.
+
+As functions, these behave as `Option.elim'`, and as an application of `some` hence the names.
```

Comment:
The distinction between `Option.elim` and `Option.elim'` is very minor. What do you think of dropping all the `'` in this file?

Reply:
Sure, I think that's fine, clearly it has the advantage of not adding to the prime linter debt.